plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/extractor/MetadataCollector.java

```diff
@@ -98,8 +108,54 @@ public Tree visit(Tree tree, ExecutionContext ctx) {
                     acc.addOtherFile(sourceFile.getSourcePath().toString());
                     LOG.debug("File {} is not a common file", sourceFile.getSourcePath());
                 }
+                groovyIsoVisitor.visit(tree, ctx);
                 return tree;
             }
+
+            final TreeVisitor<?, ExecutionContext> groovyIsoVisitor = Preconditions.check(
+                    new FindSourceFiles("**/Jenkinsfile"), new GroovyIsoVisitor<ExecutionContext>() {
+                        @Override
+                        public J.MethodInvocation visitMethodInvocation(
+                                J.MethodInvocation method, ExecutionContext ctx) {
+                            int jdkVersion = Integer.MAX_VALUE;
+                            J.MethodInvocation m = super.visitMethodInvocation(method, ctx);
+                            if ("buildPlugin".equals(m.getSimpleName())) {
+                                List<Expression> args = m.getArguments();
+
+                                for (Expression arg : args) {
+                                    if (arg instanceof G.MapEntry entry) {
+                                        if ("configurations".equals(((J.Literal) entry.getKey()).getValue())) {
+                                            if (entry.getValue() instanceof G.ListLiteral listLiteral) {
+                                                for (Expression expression : listLiteral.getElements()) {
+                                                    if (expression instanceof G.MapLiteral mapLiteral) {
+
+                                                        for (Expression mapExpr : mapLiteral.getElements()) {
+                                                            if (mapExpr instanceof G.MapEntry mapEntry) {
+
+                                                                J.Literal key = (J.Literal) mapEntry.getKey();
+                                                                J.Literal value = (J.Literal) mapEntry.getValue();
+
+                                                                if ("jdk".equals(key.getValue())) {
+                                                                    int currentJdkVersion =
+                                                                            Integer.parseInt(value.getValue()
+                                                                                    .toString());
+                                                                    if (currentJdkVersion < jdkVersion) {
```

Comment:
This logic should not be there. It's the responsibility of the user of metadata to select the JDK version. The collector should be dumb as possible